cmd/groundstation/database/interface.go

```diff
@@ -0,0 +1,15 @@
+package database
+
+import "github.com/gpuctl/gpuctl/internal/status"
+
+// define set of operations on the database that any provider will implement
+type Database interface {
+	// update the last seen time for a satellite to the current time
+	UpdateLastSeen(host string) error
+
+	// record a new data point for a satellite in the data store
+	AppendDataPoint(packet status.GPUStatusPacket) error
+
+	// get the latest metrics for all approved machines
+	LatestData() ([]status.GPUStatusPacket, error)
```

Comment:
Probably needs our schema to be defined a little bit clearer, but the latest data might be variable depending on what is requested by the front-end in terms of view on the database. It would be nice to have a way to execute santisied SQL queries against the database and return data according to the schema's data model.

Reply:
In what sort of ways do you think we might want to vary the query? I can see it being easy to add arguments to `LatestData()` so we can get the latest data for a particular machine or set of machines (shared or workstation for instance - specified by constants) - would this be enough?